python/grass/script/db.py

```diff
@@ -127,9 +127,8 @@ def db_connection(force=False, env=None):
     :return: parsed output of db.connect
     """  # noqa: E501
     try:
-        nuldev = open(os.devnull, "w")
-        conn = parse_command("db.connect", flags="g", stderr=nuldev, env=env)
-        nuldev.close()
+        with open(os.devnull, "w") as nuldev:
+            conn = parse_command("db.connect", flags="g", stderr=nuldev, env=env)
```

Comment:
[subprocess.DEVNULL](https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL) is likely a better solution.

Reply:
Personally, with how touchy the whole changes will be, I think limiting the amount of changes that affects one PR would be better. I don't see immediate benefits, and the implementation has a difference. In os.devnull, it is a string https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/os.py#L104-L105 refers eventually to https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/ntpath.py#L18 and https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/posixpath.py#L23

For subprocess.DEVNULL, it ends up being a number, used for a property with special handling throughout the file. https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/subprocess.py#L289 
https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/subprocess.py#L1144-L1147 https://github.com/python/cpython/blob/9f9b00d52ceafab6c183e8b0f502071d59dc6d22/Lib/subprocess.py#L1320-L1421

So, for it to not be "my fault" if something different breaks too much, and to be easier to see word for word the diff of before->after, do you really hold onto this change?

Reply:
We are using `subprocess.DEVNULL` in `python/grass/script/setup.py`, `python/grass/app/runtime.py` and `lib/init/grass.py`, so usage of it likely tested quite a bit. Having all replacements of `open(os.devnull...` in one PR would be one approach to limit the potential impact.

Reply:
Ok, I'll queue up that work to do in one pass at some point in a weekend.

Reply:
I didn't see much yet, GitHub's code search shows 16 results

Reply:
Most of these are benchmarks, but couple are critical code.